### ISSUE TITLE
Customizable reply routing in consumer

### ISSUE THREAD

**Issue by sashahohloma** [cody-greene/node-rabbitmq-client#36]
Hey there!
First of all, big thanks for you work! We really like your library, especially higher-level Consumer/Publisher API.

I want suggest an enhancement for Consumer API: in our application we need to route replies in specific exchanges and this is not possible due the order of Envelope properties in destructuring at line [200](https://github.com/cody-greene/node-rabbitmq-client/blob/8e3a0613e0d0dc01035f0377bf128f32a95ebcd5/src/Consumer.ts#L200) of Consumer. For now we're use monkey patch from example which allow us to config reply function like Publisher.send method and kept default behaviour from documentation.

Do you think this change would be useful for the library?

```javascript
const { Connection } = require('rabbitmq-client');
const { Consumer } = require('rabbitmq-client/lib/Consumer');

class ProcessingConsumer extends Consumer {
    _makeReplyfn(req) {
        return (body, envelope) => {
            if (!req.replyTo) {
                throw new Error('attempted to reply to a non-RPC message');
            }
            return this._ch.basicPublish({
                exchange: '',
                routingKey: req.replyTo,
                correlationId: req.correlationId,
                ...envelope,
            }, body);
        };
    }
}

class ProcessingConnection extends Connection {
    createConsumer(props, cb) {
        return new ProcessingConsumer(this, props, cb);
    }
}

exports.ProcessingConsumer = ProcessingConsumer;
exports.ProcessingConnection = ProcessingConnection;
```

**Comment by cody-greene** [comment]
Sure, seems reasonable.

-------- Original Message --------

**Comment by cody-greene** [comment]
[Fixed in 4.5.0](https://github.com/cody-greene/node-rabbitmq-client/commit/616f74bf3b840e03b8b7a250fc42065c27d7d700)
